File: lecture/dingcorithm/1st_week/01_06_find_prime_list_under_number.py
input = 20


def find_prime_list_under_number(number):
    # 이 부분을 채워보세요!
    prime_list = []
    # for i in range(2, number+1):
    #     flag = True #답이다
    #     for j in range(2, i):
    #         if i % j == 0:
    #             flag = False
    #             break
    #     if flag == True:
    #         result.append(i)

    # 소수들과만 비교하면 된다!! (현제 구하고 있었음)
    # n의 제곱근보다 크지 않은 어떤 소수로도 나누어 떨어지지 않는다
    for n in range(2, number+1):
        for i in prime_list:
            if i*i <= n and n % i == 0:
                break
        else:
            prime_list.append(n)
    return prime_list


result = find_prime_list_under_number(input)
print(result)

File: study/dingcorithm/1st_week/01_06_find_prime_list_under_number.js
// 소수는 자기 자신과 1외 에는 아무것도 나눌 수 없다.
// TODO: 다시한번 복습
function findPrimeListUnderNumber(number) {
  const primeList = [];
  for (let num = 2; num <= number; num++) {
    let isPrime = true;
    for (const i of primeList) {
      if (i * i > num && num % i === 0) {
        isPrime = false;
        break;
      }
    }
    if (isPrime) {
      primeList.push(num);
    }
  }

  return primeList;
}

const input = 20;
const result = findPrimeListUnderNumber(input);
console.log(result);
